[ADD] Field service: adding itinerary planning page (#9540)
odoo/documentation: content/applications/services/field_service/planning_itinerary.rst

=====================
Planning an Itinerary
=====================

By default, **Odoo Field Service** shows a static map where all task locations for the day are
pinned. To make it more useful for the field service workers, it is possible to display an itinerary
on the map. To do so enable the **Map Route** feature as follows.

Configure Route Map
===================

> To enable the **Map Routes feature**, a free **MapBox** account is required and needs to be linked
> to Odoo.
> 
> Go to the `Mapbox website <https://www.mapbox.com/>`_ :
>  - create an account or log in if you already have one
>  - create a token
> 
> Go to the :guilabel:`Settings` app, scroll down to :menuselection:`Integrations --> Map Routes -->
> Token --> paste your MapBox token --> Save`.

Reviewer: ```suggestion
#. Create or log in to a Mapbox account using the following link: `<https://www.mapbox.com/>`_.
#. Create a token: Click :guilabel:`Create a token`, then, in the form that opens, add a :guilabel:`Token name`, scroll down to enter your Odoo databse's :guilabel:`URL`, and enter your password.
#. Copy the access token.
#. In Odoo, open the :guilabel:`Settings` app, scroll down to :guilabel:`Integrations` section, paste your Mapbox access token in the :guilabel:`Token` field under :guilabel:`Map Routes`, and click :guilabel:`Save`.
```
- When you have several steps to perform 1 action (i.e. enable the feature here), it's better to structure it in a numbered list. 
- I would personnally explain how to create an access token since we do it for many other features such as payment providers, but we could also send people to this doc: https://docs.mapbox.com/help/getting-started/access-tokens/#adding-url-restrictions-to-access-tokens if you prefer. In any case, we need to give more info about this.
- menuselection = for menu items exclusively
- We really need to check if this feature is only used for field service, otherwise this configuration part should go in another part of the documentation

Setup your Field Service task to be featured on a map
=====================================================

> Go to :guilabel:`Field Service` and create a new task.

Reviewer: Not necessarily, right? could also be an existing task

> .. important::
>    For a Field Service task to be featured on a map, it needs:
> 
>    - a planned date
>    - a valid address linked to the customer’s profile

Reviewer: ```suggestion
.. note::
   For a Field Service task to appear on a map, it needs:

   - a planned date
   - a valid address linked to the customer’s profile
```

> Doing so sorts out the day’s field service tasks in order based on their time and date, and shows

Reviewer: doing what? what info do we want to highlight in this section?

the way from one location to the next.

Display your itinerary on a map
===============================

> Go to :menuselection:`Field Service --> My Tasks --> Map`. By default, this map shows today’s tasks.
> Remove the :guilabel:`Today` filter on the research above the map top to display all tasks. Your
> tasks are sorted by date on the left column and pinned on the map.

Reviewer: ```suggestion
To display your tasks on a map, go to :menuselection:`Field Service --> My Tasks --> Map`. 

.. tip::
   By default, the map shows today’s tasks. Remove the :guilabel:`Today` filter in the search bar to display all tasks. Your tasks are then sorted by date in the left column.
```
It seems to me that the tasks are pinned on the map in any case, even if you remove the filter.
Question: how are the tasks sorted by default for a given day? --> need to explain this before the tip, maybe?

Open your itinerary in Google Maps

> ==================================

Reviewer: I'm not sure i would put this as a title (and if it were: i'd put it as a h3) since it's part of the "Map" feature IMO so I'd just say "to open your itinerary in Google Maps, click :guilabel:`View in Google Maps`.

- How can we maybe group/filter the tasks by planned date? (might be useful to have the itinerary for tomorrow for example)
- Need to talk about how you can click a task on the map and see info + open task + navigate to

Go to :menuselection:`Field Service --> My Tasks --> Map --> View in Google Maps`

Doing so opens your itinerary on the Google Maps website or the Google Maps app if it’s installed on

> your device.

Reviewer: not needed imo

> Google Maps includes your current location as a starting point for your itinerary.

Reviewer: as a note

=====================
Planning an itinerary
=====================

By default, **Odoo Field Service** shows a static map where all task locations for the day are
pinned. To make it more useful for the field service workers, it is possible to display an itinerary
on the map. To do so enable the **Map Route** feature as follows.

Configure Route Map
===================

To enable the **Map Routes feature**, a free **MapBox** account is required and needs to be linked
to Odoo.

Go to the `Mapbox website <https://www.mapbox.com/>`_:

- create an account or log in if you already have one
- create a token

Go to the :guilabel:`Settings` app, scroll down to :menuselection:`Integrations --> Map Routes -->
Token --> paste your MapBox token --> Save`.

Setup your Field Service task to be featured on a map
=====================================================

Go to :guilabel:`Field Service` and create a new task.

> .. note::

Reviewer 2: Cher Guillaume, ceci est une suggestion, peut-être mettre en important ? Car c'est obligatoire ? A voir avec le/la next reviewer :)
